### PR TITLE
chore: update to electron 13

### PR DESCRIPTION
**Pull request by deepak1556**
- [x] Build nightly exploration builds

### CODE REVIEW

**Comment by bpasero** [comment]
Re smoke tests, make sure to merge in https://github.com/microsoft/vscode/commit/4cda850edf57f244260b7b534cbb8d7c3717f58a

**Comment by deepak1556** [comment]
They do fail after taking in the above commit, I will give this a try once internal builds are available for triggering insiders

**Comment by deepak1556** [comment]
Ready for review, plan is to merge next week to get coverage in July iteration insiders.

File: test/smoke/src/main.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ import { setup as setupDataMultirootTests } from './areas/multiroot/multiroot.te
 import { setup as setupDataLocalizationTests } from './areas/workbench/localization.test';
 import { setup as setupLaunchTests } from './areas/workbench/launch.test';
 
-const tmpDir = tmp.dirSync({ prefix: 't' }) as { name: string; removeCallback: Function; };
+const tmpDir = tmp.dirSync({ name: 't' }) as { name: string; removeCallback: Function; };
```
Comment (deepak1556) [inline test/smoke/src/main.ts:37]: Smoke tests in ci were creating socket paths longer than we expect https://github.com/microsoft/vscode/blob/5fedf0213401d8b928024151aedd589a81d9cf78/src/vs/base/parts/ipc/node/ipc.net.ts#L522-L523 , which was causing the app to quit with exception, hence this change to reduce the length of user data dir.

**Review by bpasero — APPROVED** [review]
To clarify: we only enable crash reporter now if `enable-crash-reporter` is set? Whats the idea behind this change?

**Comment by bpasero** [comment]
Weird that node.js still seems to be at `14.16.0` even though https://github.com/electron/electron/releases/tag/v13.0.0 indicates otherwise.

**Comment by deepak1556** [comment]
With E13 `submitURL` cannot be empty when `uploadToServer` is true, in smoke tests we disable crash reporter via the cli switch https://github.com/microsoft/vscode/blob/main/test/automation/src/code.ts#L143 but still end up calling `crashReporter.start`. I have cleaned up the logic by moving the deciding conditions earlier so that we make call to the crashReporter if it is disabled.

We only enable crash reporter if `enable-crash-reporter` is set or `crash-reporter-directory` argument is provided.

**Comment by deepak1556** [comment]
Yeah it is a typo in the release notes wrt node version, there is no version change in E13

**Comment by bpasero** [comment]
But doesn't that mean we have to update all our launch scripts to set `enable-crash-reporter` if we want to enable it for our users?

**Comment by deepak1556** [comment]
`enable-crash-reporter` runtime argument defaults to enabled for all our users. Also there is no change in the conditions for crash reporter enablement in this PR, I have just moved the callsites earlier to avoid unnecessary call to crashreporter API. 

Are you thinking of another case that breaks with this change ?

**Comment by bpasero** [comment]
Sorry, I totally forgot we had this in `argv.json`.

**Pull request merged**